Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2412,7 +2412,18 @@ static String preferredFilenameForElement(const HTMLImageElement& element)
     return wrapper;
 }
 
-static RetainPtr<NSAttributedString> attributedStringWithAttachmentForElement(const HTMLImageElement& element)
+template<typename Element>
```

Comment:
Why is `template<typename Element>` needed?

Reply:
because it can accept both an HTMLImageElement or an HTMLAttachmentElement

Reply:
Both are `Element` subclasses though.

Reply:
that won't be sufficient, because of the call to `fileWrapperForElement`, which is an overloaded method which requires either of `HTMLImageElement` or `HTMLAttachmentElement`. So `attributedStringWithAttachmentForElement` needs specifically either of those types, and not any subclass of Element

Reply:
I see... slightly confusing to use `typename Element` given we have a very commonly used class called `Element` – but I guess it's fine...

Reply:
refactored

Reply:
Yeah, I _think_ we generally use something like `ElementType` here to avoid confusion.